### DISCUSSION TITLE
Compact historical data in different granularity

### DISCUSSION THREAD

**Discussion by ylin30** [ytyou/ticktock#41] [Ideas]
- store data in full granularity (~10s interval) for about 3 months
- store data aggregated to e.g. min, max, avg (~5min interval) for one year
- store data aggregated to e.g. min, max, avg (1d interval) forever

This feature is proposed by @jens-ylja based on [the original discussion #40 ](https://github.com/ytyou/ticktock/discussions/40)

**Comment by juggledad** [comment]
I like this idea!

**Comment by ylin30** [comment]
FYI. We are implementing a slightly different strategy into a future release v0.20 right now. 

- We will keep all the original data forever. So users never lose the details of the original data.

Since TT's data files are stored in folders with timestamps in names, users can easily delete old data files at any time if they decide not to keep the original data.

In the meanwhile, we provide aggregated data of min/max/sum/count (default 1 hour interval) to accelerate queries. 

- If a query range is long enough, we will use aggregated data. It will be much faster. 
- On the other hand, if a query range is short or the query specially wants to zoom into small granularity of data (i.e., smaller downsampling than 1 hour), we will use original data.
